File: algorithms/recursive/euclidean-algorithm/euclidean-algorithm.ts
function euclideanAlgorithm(a: number, b: number): number {
  if (b === 0) {
    return a
  }

  return euclideanAlgorithm(b, a % b)
}

console.log(euclideanAlgorithm(1, 10))
console.log(euclideanAlgorithm(56, 98))
console.log(euclideanAlgorithm(105, 252))

File: algorithms/recursive/euclidean-algorithm/euclidean-algorithm.go
package main

import "fmt"

func euclideanAlgorithm(a, b int) int {
	if b == 0 {
		return a
	}

	return euclideanAlgorithm(b, a%b)
}

func main() {
	fmt.Println(euclideanAlgorithm(1, 10))
	fmt.Println(euclideanAlgorithm(56, 98))
	fmt.Println(euclideanAlgorithm(105, 252))
}
